src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -15,9 +15,15 @@ function fill_open_boundary_regions!(field, boundary_conditions, indices, loc, g
     # gets `open_fill`, the function which fills open boundaries at `loc`, as well as `regular_fill`
     # which is the function which fills non-open boundaries at `loc` which informs `fill_halo_size` 
     open_fill, regular_fill = get_open_halo_filling_functions(loc) 
-    fill_size = fill_halo_size(field, regular_fill, indices, boundary_conditions, loc, grid)
 
-    launch!(arch, grid, fill_size, open_fill, field, left_bc, right_bc, loc, grid, args)
+    # Computing the size of the fill halo kernel. We use the `regular_fill` function to
+    # calculate the index range. Remember: windowed fields require the halos
+    # to be computed only on a part of the boundary so we need to offset the kernel indices 
```

Comment:
I don't understand how we "use a function" to calculate an index range. This doesn't make sense to me. The index range depends on indices, loc, boundary conditions, grid? Why don't we use those arguments to compute the index range?

Reply:
This has been resolved in https://github.com/CliMA/Oceananigans.jl/pull/3792

Reply:
Touche, I thought it was just me that has a weird dream where I was a broken record going round and round...